### PR TITLE
[SYCL][FPGA][NFC] Tidy up intel_fpga_reg codegen test

### PR DESCRIPTION
**Pull request by mlychkov**
Checks for Clang::CodeGenSYCL/intel-fpga-reg.cpp test were autogenerated
to speed up fixing of LIT tests failures after a patch
"[SYCL] Refactor address space handling in CodeGen library (#2864)".

Reorganize test code and tidy up checks to improve readability.

Signed-off-by: Mikhail Lychkov <mikhail.lychkov@intel.com>

### CODE REVIEW

**Review by MrSidims — APPROVED** [review]

**Review by erichkeane — DISMISSED** [review]
Changes look sane to me.  I just did a spot-check, so someone else might need to poke at the check-lines.

**Review by bader — DISMISSED** [review]

File: clang/test/CodeGenSYCL/intel-fpga-reg.cpp [hunk 1]
```diff
@@ -22,273 +23,166 @@ class A {
...
+  // CHECK-NEXT: [[V_AB5:%.*]] = call i32 @llvm.annotation.i32(i32 [[V_AB4]], [[BIFR_STR]]
+  // CHECK-NEXT: store i32 [[V_AB5]], i32 addrspace(4)* %e
+
   int f;
   f = __builtin_intel_fpga_reg(a);
+  // CHECK: [[V_F1:%.*]] = load i32, i32 addrspace(4)* %a
+  // CHECK-NEXT: [[V_F2:%.*]] = call i32 @llvm.annotation.i32(i32 [[V_F1]], [[BIFR_STR]]
+  // CHECK-NEXT: store i32 [[V_F2]], i32 addrspace(4)* %f
+
+  myInt myA = 321;
```
[the remark this answers is not in the document]
Comment (mlychkov) [reply clang/test/CodeGenSYCL/intel-fpga-reg.cpp:66]: Technically there is no difference between them. I may remove this case as duplicate or change a type to cover a case with the other type.

[the remark this answers is not in the document]
Comment (elizabethandrews) [reply clang/test/CodeGenSYCL/intel-fpga-reg.cpp:66]: I think you can just remove it.

[the remark this answers is not in the document]
Comment (mlychkov) [reply clang/test/CodeGenSYCL/intel-fpga-reg.cpp]: Done.

**Review by MrSidims — APPROVED** [review]

**Review by elizabethandrews — APPROVED** [review]

**Pull request merged**